exporter/clickhousetracesexporter/clickhouse_exporter_v3.go

```diff
@@ -0,0 +1,324 @@
+			statusString, err := strconv.Atoi(v.Str())
+			statusInt := v.Int()
+			if err == nil && statusString != 0 {
+				statusInt = int64(statusString)
+			}
+			span.ResponseStatusCode = strconv.FormatInt(statusInt, 10)
+		} else if k == "rpc.jsonrpc.error_code" {
+			span.ResponseStatusCode = v.Str()
+		}
+		return true
+
+	})
+
+}
+
+func populateEventsV3(events ptrace.SpanEventSlice, span *SpanV3, lowCardinalExceptionGrouping bool) {
+	for i := 0; i < events.Len(); i++ {
+		event := Event{}
+		event.Name = events.At(i).Name()
+		event.TimeUnixNano = uint64(events.At(i).Timestamp())
+		event.AttributeMap = map[string]string{}
+		event.IsError = false
+		events.At(i).Attributes().Range(func(k string, v pcommon.Value) bool {
+			event.AttributeMap[k] = v.AsString()
+			return true
+		})
+		if event.Name == "exception" {
+			event.IsError = true
+			span.ErrorEvent = event
+			uuidWithHyphen := uuid.New()
+			uuid := strings.Replace(uuidWithHyphen.String(), "-", "", -1)
+			span.ErrorID = uuid
+			var hash [16]byte
+			if lowCardinalExceptionGrouping {
+				hash = md5.Sum([]byte(span.ServiceName + span.ErrorEvent.AttributeMap["exception.type"]))
+			} else {
+				hash = md5.Sum([]byte(span.ServiceName + span.ErrorEvent.AttributeMap["exception.type"] + span.ErrorEvent.AttributeMap["exception.message"]))
+
+			}
+			span.ErrorGroupID = fmt.Sprintf("%x", hash)
+		}
+		stringEvent, _ := json.Marshal(event)
+		span.Events = append(span.Events, string(stringEvent))
+	}
+}
+
+func newStructuredSpanV3(bucketStart uint64, fingerprint string, otelSpan ptrace.Span, ServiceName string, resource pcommon.Resource, config storageConfig) (*SpanV3, error) {
+	durationNano := uint64(otelSpan.EndTimestamp() - otelSpan.StartTimestamp())
+
+	isRemote := "unknown"
+	flags := otelSpan.Flags()
+	if flags&hasIsRemoteMask != 0 {
+		isRemote = "no"
+		if flags&isRemoteMask != 0 {
+			isRemote = "yes"
+		}
+	}
+
+	attributes := otelSpan.Attributes()
+	resourceAttributes := resource.Attributes()
+	tagMap := map[string]string{}
+	stringTagMap := map[string]string{}
+	numberTagMap := map[string]float64{}
+	boolTagMap := map[string]bool{}
+	spanAttributes := []SpanAttribute{}
+
+	resourceAttrs := map[string]string{}
+
+	attributes.Range(func(k string, v pcommon.Value) bool {
+		tagMap[k] = v.AsString()
+		spanAttribute := SpanAttribute{
+			Key:      k,
+			TagType:  "tag",
+			IsColumn: false,
+		}
+		if v.Type() == pcommon.ValueTypeDouble {
+			numberTagMap[k] = v.Double()
+			spanAttribute.NumberValue = v.Double()
+			spanAttribute.DataType = "float64"
+		} else if v.Type() == pcommon.ValueTypeInt {
+			numberTagMap[k] = float64(v.Int())
+			spanAttribute.NumberValue = float64(v.Int())
+			spanAttribute.DataType = "float64"
+		} else if v.Type() == pcommon.ValueTypeBool {
+			boolTagMap[k] = v.Bool()
+			spanAttribute.DataType = "bool"
+		} else {
+			stringTagMap[k] = v.AsString()
+			spanAttribute.StringValue = v.AsString()
+			spanAttribute.DataType = "string"
+		}
```

Comment:
My one issue with this is that it makes the maps unusable. For example, until recently, the clickhouse system table receiver sent a map in the attributes called `ProfilesEvents` with map values with keys and values. This is completely unusable from the UI. There is no point in storing if they can't be used from query builder. I believe we should flatten them here.
![Screenshot 2024-11-10 at 6 50 23 PM](https://github.com/user-attachments/assets/891e143e-69b7-4950-a252-f8af3451a0eb)

Reply:
Because the logs exporter was making them unusable strings. I had to flatten at the source https://github.com/SigNoz/signoz-otel-collector/pull/442. Ideally, the logs exporter itself should have flattened them for me.

Reply:
Something like this is good https://go.dev/play/p/XVvXYASYFnu ?

Reply:
Yes, this looks good to me. Hopefully, the ClickHouse JSON type will soon be stable enough for use in such cases. Meanwhile, this is 100x better than the current unusable string.

Reply:
Sure let me add it.

Reply:
I have added the support for it, but not able to test it right now. Seems like the SDK's don't support maps as of now

```
Invalid type dict for attribute 'mydata' value. Expected one of ['bool', 'str', 'bytes', 'int', 'float'] or a sequence of those types
```

Reply:
Yep, added.